### PR TITLE
feat: Add code compatibility with landing on stars.

### PR DESCRIPTION
**Pull request by Codetoil**
This is the PR I talked about [Galacticraft-API#27](https://github.com/TeamGalacticraft/GalacticraftAPI/pull/27). 

> I'd argue that even though landable stars are outside the scope of the galacticraft itself, It would still be a good idea to add them to the api, just for the fact that they might be implemented in add-ons. For GC4, it was impossible to make stars landable without [patching the code](https://github.com/Codetoil/ToTheStars), so I thought the same would be the case for GC5. It doesn't look like that's correct, but regardless, I plan to also create a similar PR but for GC4. Maybe if it gets added to GC4 you'll consider adding it to GC5?

As always, feedback?

### CODE REVIEW

**Comment by Codetoil** [comment]
@marcus8448 New PR!

File: .editorconfig [hunk 1]
```diff
@@ -6,9 +6,9 @@ insert_final_newline = true
 
 [*.java]
 charset = utf-8
-indent_style = tab
+indent_style = space # 4 spaces, remember!
```
Comment (JustPyrrha) [inline .editorconfig:9]: Out of scope for PR, and especially don't mess with editorconfig since it affects literally the entire project. I assume the "4 spaces" you're referencing are from the GC5 contributing guidelines, don't mix projects like that.

File: build.gradle [hunk 2]
```diff
@@ -18,7 +18,7 @@ java.toolchain.languageVersion = JavaLanguageVersion.of(8)
 group = "micdoodle8.mods.galacticraft"
 archivesBaseName = "Galacticraft"
 
-version = '4.0.4-SNAPSHOT'
+version = '4.0.4-tts-SNAPSHOT'
```
Comment (ROMVoid95) [inline build.gradle:21]: All the changes here are out of scope for PR, nor do they fix any issues it may have, If you must change a build file in a project locally, ensure it is not commited.

File: gradle/resources.gradle [hunk 3]
```diff
@@ -1,8 +1,8 @@
 def resourceTargets = [ 'mcmod.info' ]
```
Comment (ROMVoid95) [inline gradle/resources.gradle:1]: the comment in `build.gradle` applies here as well

File: gradle/resources.gradle [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
-def intoTargets = [ 
+def intoTargets = [
```
Comment (ROMVoid95) [inline gradle/resources.gradle:2]: the comment in build.gradle applies here as well

**Comment by Codetoil** [comment]
@ROMVoid95 Some of the changes were just automatic things done by the editor.

**Comment by JustPyrrha** [comment]
> @ROMVoid95 Some of the changes were just automatic things done by the editor.

then dont check them in when you commit

**Comment by Codetoil** [comment]
> > @ROMVoid95 Some of the changes were just automatic things done by the editor.
> 
> then dont check them in when you commit

I didn't even notice the changes even occurred.

**Comment by ROMVoid95** [comment]
I'm going to have to uphold the reasoning for not accepting this PR made in https://github.com/TeamGalacticraft/GalacticraftAPI/pull/27. With emphisis on the fact this API is currently depended on by other addons and mods. 

Another problem, this would directly affect systems added by addons which GC should not be doing. As doing so can possibly introduce bugs and/or runtime isssues down the line.

However, I have been wanting to add the capability to register custom celestial types directly to the GC Registry instead of only being constrained to extending the current types GC defines. Since this would also be a feature that would benefit what your wanting to do here. and would be willing to implement this, I would be more than happy to further review a PR for that.
